packages/contracts/contracts/foundry/SphinxUtils.sol

```diff
@@ -1038,27 +1036,38 @@ contract SphinxUtils is SphinxConstants, StdUtils {
     }
 
     function fetchNumCreateAccesses(
-        Vm.AccountAccess[] memory _accesses
+        Vm.AccountAccess[] memory _accesses,
+        uint256 _chainId
     ) public pure returns (uint) {
         uint numCreateAccesses = 0;
         for (uint i = 0; i < _accesses.length; i++) {
-            if (_accesses[i].kind == VmSafe.AccountAccessKind.Create) {
+            if (isCreateAccountAccess(_accesses[i], _chainId)) {
                 numCreateAccesses += 1;
             }
         }
         return numCreateAccesses;
     }
 
+    function isCreateAccountAccess(
+        Vm.AccountAccess memory _access,
+        uint256 _chainId
+    ) private pure returns (bool) {
+        return
+            _access.kind == VmSafe.AccountAccessKind.Create &&
+            _access.chainInfo.chainId == _chainId;
+    }
+
     function fetchDeployedContractSizes(
-        Vm.AccountAccess[] memory _accesses
+        Vm.AccountAccess[] memory _accesses,
+        uint256 _chainId
```

Comment:
We're going to remove this function and `fetchNumCreateAccesses` when we remove the Moonbeam heuristic, so I didn't add additional tests for them despite slightly increasing their complexity